packages/itwin/reports-config-widget/src/widget/components/AddMappingsModal.tsx

```diff
@@ -43,35 +44,36 @@ const fetchMappings = async (
   }
 };
 
-interface AddMappingsModalProps {
+export interface AddMappingsModalProps {
   reportId: string;
   existingMappings: ReportMappingAndMapping[];
   show: boolean;
-  returnFn: () => Promise<void>;
+  onClose: () => Promise<void>;
 }
 
 export const AddMappingsModal = ({
   reportId,
   existingMappings,
   show,
-  returnFn,
+  onClose,
 }: AddMappingsModalProps) => {
   const [isLoading, setIsLoading] = useState<boolean>(false);
-  const [selectedMappings, setSelectedMappings] = useState<Mapping[]>([]);
+  const selectedMappings = useRef<Mapping[]>([]);
```

Comment:
Why the change from state to ref?

Reply:
For now, I am no longer using the useState to render the selected mappings. Just to store them. It will save some unnecessary re-renders.

Reply:
So how does the UI know what mappings are selected? I think we might encounter hard-to-debug bugs related to this.

Reply:
The table itself keeps track of this.